### DISCUSSION TITLE
Changed git hashs after sync with compress intended?

### DISCUSSION THREAD

**Discussion by erik-rw** [git-town/git-town#4314] [Q&A] [answered]
Using the compress sync feature strategy. The docs say

> When using the compress sync strategy, [git town sync](https://www.git-town.com/commands/sync) first merges the tracking and parent branches and then [compresses](https://www.git-town.com/commands/compress) the synced branch.

which git town does perfectly. 

But if there are no changes (not locally or remotely), the existing commit is still replaced with a new one. This triggers another (unnecessary) CI run.

I wonder if this is expected behavior or if there should be a way to avoid the recreation.

**Comment by kevgo** [comment, marked as answer]
Definitely not intended. This behavior is a side effect of the current (simple) implementation.

**Reply by erik-rw** [reply]
Thx for the feedback. I lack the time to look into this myself.
If others are struggling with this, too, we could create an issue for it.

For now, I will work around it by keeping most child branches as prototype branches. If CI is unaware of recreated commits, it does not rerun them.

**Reply by stephenwade** [reply]
I've created #4342 to track this.

**Comment by kevgo** [comment]
This is now available as part of [Git Town 20.2](https://github.com/git-town/git-town/releases/tag/v20.2.0).
